examples/sample_measurement/_array_utils.py

```diff
@@ -0,0 +1,50 @@
+"""Double2DArray Conversion Utilities."""
+
+from typing import Any
+
+from ni_measurement_plugin_sdk_service._internal.stubs.ni.protobuf.types import (
+    array_pb2,
+)
+
+try:
+    import numpy as np
+    from numpy.typing import NDArray
+
+    def double2darray_to_ndarray(double2darray: array_pb2.Double2DArray) -> NDArray[np.float64]:
```

Comment:
I have tested the `numpy` version as well. Also, I have added the copy and tests as per the suggestion, but in doing so I had to remove the type `NDArray[np.flost64]` and use `Any` in the method signature to keep `_array_utils` consistent accross `sample_measurement` and `service`.

This was done because `mypy` was throwing error about method signature not being correct and it was different in case of sample_measurement where numpy was not available and in service where numpy was available.

Reply:
There are some issues that is coming up to keep the `_array_utils` consistent across `sample_measurement` and `service`. 

Since, `sample_measurement` doesn't have `numpy` package. In the except block, I can't use the `NDArray[np.float64]` type def and I need to keep it to `Any`. But since, `service` contains `numpy` I have to keep the type def in the except block as `NDArray[np.float64]` otherwise mypy throws error regarding incorrect method signature.

> error: All conditional function variants must have identical signatures  [misc]
> note: Original:
> note:     def double2darray_to_ndarray(double2darray: Double2DArray) -> ndarray[Any, dtype[floating[_64Bit]]]
> note: Redefinition:
> note:     def double2darray_to_ndarray(double2darray: Double2DArray) -> Any

Reply:
One Solution would be to just remove the type def for NDArray and keep it as Any in all places where NDArray[np.float64] is used but I'm curious if then we would have to handle cases where NDArray is not being passed to the method.

Reply:
Ok. Do it like this: https://github.com/ni/measurement-plugin-python/blob/main/packages/service/ni_measurement_plugin_sdk_service/session_management/_reservation.py

1. `from __future__ import annotations` prevents type hints from being evaluated at run time.
2. Import numpy at file scope only if `typing.TYPE_CHECKING` is true.
3. Import numpy again in each function that needs it. If numpy is not installed, this will raise an `ImportError`.

```
from __future__ import annotations
...
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import numpy as np
    import numpy.typing as npt

def double2darray_to_ndarray(double2darray: array_pb2.Double2DArray) -> npt.NDArray[np.float64]:
    """Convert Double2DArray to numpy NDArray."""
    import numpy as np
    return np.array(double2darray.data, dtype=np.float64).reshape(
        double2darray.rows, double2darray.columns
    )
```

Reply:
Added this change in latest commit.